src/Microsoft.Azure.WebJobs.Host/Executors/TriggeredFunctionExecutor.cs

```diff
@@ -30,16 +31,46 @@ public FunctionDescriptor Function
             }
         }
 
-        public async Task<FunctionResult> TryExecuteAsync(TriggeredFunctionData input, CancellationToken cancellationToken)
+        public Task<FunctionResult> TryExecuteAsync(TriggeredFunctionData input, CancellationToken cancellationToken)
+        {
+            return TryExecuteAsync(input, cancellationToken, null);
+        }
+
+        public async Task<FunctionResult> TryExecuteAsync(TriggeredFunctionData input, CancellationToken cancellationToken, Func<Func<Task>, Task> hook)
         {
             IFunctionInstance instance = _instanceFactory.Create((TTriggerValue)input.TriggerValue, input.ParentId);
+            if (hook != null)
+            {
+                IFunctionInvoker invoker = new InvokeWrapper(instance.Invoker, hook);
+                instance = new FunctionInstance(instance.Id, instance.ParentId, instance.Reason, instance.BindingSource, invoker, instance.FunctionDescriptor);
```

Comment:
Why not? This doesn't need any factory services. What did you have in mind? An alternative approaches could be to make Invoker a mutable property  and just change it on the existing instance.